### PR TITLE
Add RxUIMenu

### PR DESCRIPTION
**Pull request by ssvetina**
# Description

This PR contains adding a UIMenu component to the catalog. An example button has been added. Please see video for further reference:

https://user-images.githubusercontent.com/32631956/224997283-fc8c0ecd-2398-46f6-beef-784d01c3c8c5.mp4

# Checklist:

- [x] Have checked there is no same component already in catalog.
- [x] Have created a sufficient example or wrote tests for it.
- [x] Code is structured, well written using standard Swift coding style.
- [x] All public methods and properties have meaningful and concise documentation.
- [x] Used `public` modifier for all method and properties that could be changed from user of your feature, and `private` for internal properties.
- [x] Removed any reference to the project that piece of code was created in.
- [x] Reduced the dependencies to minimum.

### CODE REVIEW

File: Catalog/Examples/CatalogDataSource.swift [hunk 1]
```diff
@@ -44,15 +44,29 @@ private extension CatalogDataSource {
...
+                RatioTransitionViewController.self,
+                RoundCornersViewController.self,
+                UIViewModifiersViewController.self,
+                ImageFromColorViewController.self,
+                CatalogNavigationController.self,
+                ToggleViewController.self,
+                LineHeightViewController.self
+            ]
+        }
+
```
Comment (nikolamajcen) [inline Catalog/Examples/CatalogDataSource.swift:69]: The same goes for this one as well.

File: Catalog/Examples/UI/RxUIMenu/RxUIMenu.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+/// A container for grouping related menu elements in an app menu or contextual menu.
+///
+/// Create RxUIMenu wrapper around the UIMenu objects and use them to construct the menus and submenus your app displays.
+/// Used in case where reactive action is needed.
+@available(iOS 15.0, *)
+class RxUIMenu<T> {
+
+    let title: String
+    let image: UIImage?
+    let children: [RxUIAction<T>]
```
Comment (nikolamajcen) [inline Catalog/Examples/UI/RxUIMenu/RxUIMenu.swift:24]: Can children be a non-action? If not, I would go with `actions` instead.

**Review by nikolamajcen — APPROVED** [review]
In general - all good, but revisit the comments.

File: Catalog/Examples/UI/RxUIMenu/RxUIMenu.swift (hunk 2 above)
Comment (ssvetina) [reply Catalog/Examples/UI/RxUIMenu/RxUIMenu.swift:24]: This is named after Apple's own naming on UIMenu so it kind of made sense to keep it this way :)

**Pull request merged**